### PR TITLE
Using `Option` on FFI

### PR DESCRIPTION
**Pull request by robertbastian**
This is ABI compatible but API breaking in C++ and arguably JS.

### CODE REVIEW

**Review by Manishearth — CHANGES_REQUESTED** [review]
we can't land this until 2.0

**Comment by robertbastian** [comment]
Do you consider this breaking in JS?

I'm happy to change the CPP1 generator to translate `Option` to `diplomat::Result<T, std::monostate>` to not have a breaking change there.

**Comment by Manishearth** [comment]
I'm personally ok with breaking JS. I'll leave it up to @sffc

File: ffi/capi/bindings/js/ICU4XScriptExtensionsSet.d.ts
Comment (sffc) [inline ffi/capi/bindings/js/ICU4XScriptExtensionsSet.d.ts:33]: Issue: the implementation returns null not undefined

**Review by sffc — APPROVED** [review]
The JS change is breaking but the TS should help and I think we don't have very many JS users and the impacted functions are sufficiently in the corner that I'm okay with a changelog notice

File: ffi/capi/bindings/js/ICU4XScriptExtensionsSet.d.ts
Comment (robertbastian) [reply ffi/capi/bindings/js/ICU4XScriptExtensionsSet.d.ts:33]: ah that's a bug in Diplomat then

**Review by Manishearth — APPROVED** [review]

**Pull request merged**
